tests/infra_test.go

```diff
@@ -688,14 +690,15 @@ var _ = Describe("[Serial]Infrastructure", func() {
 					} else {
 						resp.Body.Close()
 					}
-					errors <- err
+					errorsChan <- err
 				}(ix)
 			}
 
 			errorCount := 0
 			for ix := 0; ix < concurrency; ix++ {
-				err := <-errors
+				err := <-errorsChan
 				if err != nil {
+					Expect(checkError(err)).To(Succeed())
```

Comment:
This reads as: I expect that the error check succeeds.
What is checked is not clear so this is not a fit.

IMO you should wrap the whole for-loop, return an error (not assert) and at line 707 assert on it (Succeed()).
e.g. `Expect(waitHttpResponse(errorsChan).To(Succeed())`

If the number of error count is important, move it into the helper function (`waitHttpResponse`) and print it there before returning.

Please see also the problem with the goroutine leak here (prev comment).

Reply:
so i will just change it to BeNil , i prefer it as this if possible
i perfer the logic that iterate on the loop in the test itself, no need to send all the parameters to another level

Reply:
The logic cannot be read, which makes it improper from my perspective.
I also do not understand why you think that exposing all this details is correct or better. When you cannot read it without getting into the details of the inner function, then it is not abstracting it correctly.

Reply:
deleted before saw you answered, because i am changing it, even that i dont agree its not readable.